<p>
    This rule checks that the batch layer of one component does not depend on the logic layer of another component.
    Allowed dependencies from the batch layer:
    <ul type='disc'>
        <li>Batch layer of a different component</li>
        <li>Logic layer of the same component</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>compA.batch</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>compB.logic</del></span>.api.Bar;

    public class MyClass {}
</pre>